galter_subjects_utils/deltor.py

```diff
@@ -0,0 +1,210 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2024 Northwestern University.
+#
+# galter-subjects-utils is free software; you can redistribute it and/or
+# modify it under the terms of the MIT License; see LICENSE file for more
+# details.
+
+"""Generate delta operations."""
+
+from dataclasses import dataclass
+
+
+@dataclass
+class Analysis:
+    """
+    Holds results of analyzing a src subject.
+
+    Can't be a namedtuple because it needs to be mutable.
+    """
+
+    id: str
+    seen: bool = False  # present in dst subjects
+    relabelled: str = ""  # new label
+    replaced: str = ""  # id of replacing subject
+
+
+class DeltasGenerator:
```

Comment:
Is there any reason you named the file "deltor" instead of "deltas_generator"? I guess the likely reason is brevity. "delta generator" to "deltor" took me a second, but it's growing on me.

Reply:
Yes brevity and consistency with other module names: they were all "doers" without subject/object {pre,suf}-fixes, so I kept it the same (I tried "delter.py" but that was too far from deltas generator). I am running 2 "naming experiments" to see if they improve readability: 

- unabashedly coming with contracted neologisms to put a pithy name to complex entities (case in point this name) 
- using variable names where the main entity is the first part of the name (e.g. `subjects_replaced` instead of `replaced_subjects`). I am finding code understanding much smoother this way so far, despite the word order not matching English.